testing/upgradetest/upgrader.go

```diff
@@ -507,6 +513,12 @@ func WaitHealthyAndVersion(ctx context.Context, f *atesting.Fixture, versionInfo
 			return ctx.Err()
 		case <-t.C:
 			err := CheckHealthyAndVersion(ctx, f, versionInfo)
+			// If we're in an upgrade process, the versions might not match
+			// so we wait to see if we get to a stable version
+			if errors.Is(err, ErrVerMismatch) {
```

Comment:
The one thing I'm not 100% sure is if this could slow down some other tests that now will need to wait for the timeout. If that's a problem, I can add an "IgnoreVersionMismatch" flag.